From 3299fd3aa9aaad11bd626cde60c9c50beef53bdb Mon Sep 17 00:00:00 2001
From: proudzhu <proudzhu.fdu@gmail.com>
Date: Sun, 18 Oct 2015 01:17:14 +0800
Subject: [PATCH] fix "undefined symbol: avcodec_free_frame
 avcodec_alloc_frame" with recent ffmpeg

the two functions is removed in ffmpeg commit
(8f12ef9860d0e164e4647fd5d5cebdb3cfb34a79)
---
 plugins/ffmpeg/ffmpeg.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/plugins/ffmpeg/ffmpeg.c b/plugins/ffmpeg/ffmpeg.c
index 881c3f466..c85ecf9d7 100644
--- a/plugins/ffmpeg/ffmpeg.c
+++ b/plugins/ffmpeg/ffmpeg.c
@@ -242,7 +242,9 @@ ffmpeg_init (DB_fileinfo_t *_info, DB_playItem_t *it) {
     memset (&info->pkt, 0, sizeof (info->pkt));
     info->have_packet = 0;
 
-#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(53, 40, 0)
+#if LIBAVCODEC_VERSION_INT > AV_VERSION_INT(55, 28, 0)
+    info->frame = av_frame_alloc();
+#elif LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(53, 40, 0)
     info->frame = avcodec_alloc_frame();
 #endif
 
@@ -286,7 +288,11 @@ ffmpeg_free (DB_fileinfo_t *_info) {
     trace ("ffmpeg: free\n");
     ffmpeg_info_t *info = (ffmpeg_info_t*)_info;
     if (info) {
-#if LIBAVCODEC_VERSION_INT > AV_VERSION_INT(54, 59, 100)
+#if LIBAVCODEC_VERSION_INT > AV_VERSION_INT(55, 28, 0)
+        if (info->frame) {
+            av_frame_free(&info->frame);
+        }
+#elif LIBAVCODEC_VERSION_INT > AV_VERSION_INT(54, 59, 100)
         if (info->frame) {
             avcodec_free_frame(&info->frame);
         }
